### ISSUE TITLE
add more metadata (dandiset id, command) into log filename name?

### ISSUE THREAD

**Issue by yarikoptic** [dandi/dandisets#238]
in one of the recent PRs logging was RFed to no longer store logs within any particular dandiset. So we have them all in a single location within superdataset:

```
(base) dandi@drogon:/mnt/backup/dandi/dandisets/.git/dandi/backups2datalad$ ls -lt | head
total 361346464
-rw-r--r-- 1 dandi dandi  86346599315 Jul 28 18:30 2022.07.18.21.04.47Z.log
-rw-r--r-- 1 dandi dandi       211328 Jul 28 18:26 2022.07.18.19.39.18Z.log
-rw-r--r-- 1 dandi dandi     15722179 Jul 28 18:21 2022.07.28.21.59.07Z.log
-rw-r--r-- 1 dandi dandi     15848460 Jul 28 16:22 2022.07.28.19.57.24Z.log
...
```
I thought it would be fine but it is pretty much is impossible now to identify which log file to look at for a particular dandiset of interest (e.g. 000026 for #230 ). Moreover it is complicated by the fact that most recent log for a long running task of updating 000108 is already >80GB in size making "grep"ing non trivial.  I ended up trying smth like

```
ls -t | while read f; do head -n 1000 $f | grep -q 000026 && { echo $f; break; } || continue; done
```
to find first recent file mentioning 000026

We need to come up with some way to ease our identification of which dandiset or zarr file any particular log is about.  Do you see a way @jwodder ?

next problem -- I found that file mentioning 000026 but it was for populate* command , not `update-from-backup` and `populate` was not even grepping within that file.  If we could include command within log file name -- great. If not -- let's at least log the invocation so we could grep for it.

**Comment by jwodder** [comment]
@yarikoptic 

> We need to come up with some way to ease our identification of which dandiset or zarr file any particular log is about.

The logs in that folder are for the entirety of a backups2datalad run, so they cover multiple Dandisets (unless you specified just one specific Dandiset as part of the command) and multiple Zarrs.

> If we could include command within log file name -- great

That can be done.

**Comment by yarikoptic** [comment]
> The logs in that folder are for the entirety of a backups2datalad run, so they cover multiple Dandisets (unless you specified just one specific Dandiset as part of the command) and multiple Zarrs.

since we process them in parallel, we might need to add logging dandiset id or zarr id in each line then to tell one "process" from another.  Do you think this could be done reasonably easily?

**Comment by jwodder** [comment]
@yarikoptic The logs already include the Dandiset ID and Zarr ID in each line.  The only exceptions are log messages from external libraries and the log messages emitted whenever an external command is run (but at least the latter include the CWD).

**Comment by yarikoptic** [comment]
I am not observing that generally (some lines include API urls pointing to specific dandisets indeed) to be true:

```shell
(dandi-cli-latest) dandi@drogon:/mnt/backup/dandi/dandisets/.git/dandi/backups2datalad$ grep -v 000 2022.07.29.14.33.54Z.log | sed -e 's,\],,g' | uniq -c | head -n 20
      1 2022-07-29T10:33:54-0400 [INFO     backups2datalad: Saving logs to /mnt/backup/dandi/dandisets/.git/dandi/backups2datalad/2022.07.29.14.33.54Z.log
      1 2022-07-29T10:34:06-0400 [DEBUG    httpx._client: HTTP Request: GET https://api.dandiarchive.org/api/dandisets/ "HTTP/1.1 200 OK"
      1 2022-07-29T10:34:06-0400 [DEBUG    urllib3.connectionpool: Starting new HTTPS connection (1): rig.mit.edu:443
      1 2022-07-29T10:34:07-0400 [DEBUG    urllib3.connectionpool: https://rig.mit.edu:443 "GET /et/projects/dandi/dandi-cli HTTP/1.1" 200 579
      1 2022-07-29T10:34:07-0400 [WARNING  dandi: A newer version (0.45.1) of dandi/dandi-cli is available. You are using 0.40.0
      2 2022-07-29T10:34:08-0400 [INFO     backups2datalad: Updating metadata file
      1 2022-07-29T10:34:08-0400 [DEBUG    httpx._client: HTTP Request: GET https://api.dandiarchive.org/api/dandisets/?page=2 "HTTP/1.1 200 OK"
      3 2022-07-29T10:34:08-0400 [INFO     backups2datalad: Updating metadata file
      1 2022-07-29T10:34:13-0400 [DEBUG    backups2datalad: Version 0.210812.1448 already tagged
      1 2022-07-29T10:34:13-0400 [DEBUG    backups2datalad: Version 0.220126.1903 already tagged
      1 2022-07-29T10:34:13-0400 [DEBUG    backups2datalad: Version 0.220126.1851 already tagged
      1 2022-07-29T10:34:13-0400 [DEBUG    backups2datalad: Version 0.220126.1855 already tagged
      1 2022-07-29T10:34:13-0400 [DEBUG    backups2datalad: Version 0.220126.1852 already tagged
      1 2022-07-29T10:34:14-0400 [DEBUG    backups2datalad: Version 0.220126.1853 already tagged
      3 2022-07-29T10:34:22-0400 [INFO     backups2datalad: Updating metadata file
      2 2022-07-29T10:34:23-0400 [INFO     backups2datalad: Updating metadata file
      1 2022-07-29T10:34:28-0400 [DEBUG    backups2datalad: Version 0.220126.1907 already tagged
      1 2022-07-29T10:34:28-0400 [DEBUG    backups2datalad: Version 0.220126.1905 already tagged
      1 2022-07-29T10:34:28-0400 [DEBUG    backups2datalad: Version 0.220126.1903 already tagged
      1 2022-07-29T10:34:42-0400 [INFO     backups2datalad: Updating metadata file
```

so it is not clear to what dandiset those actions (updating metadata, versions) are pertinent to

**Comment by jwodder** [comment]
@yarikoptic #240 should fix that.

**Comment by yarikoptic** [comment]
lets assume it was fixed by  #240 for now
